ESP32LapTimer/src/CrashDetection.cpp

```diff
@@ -0,0 +1,29 @@
+#include "CrashDetection.h"
+
+#include <rom/rtc.h>
+#include <Arduino.h>
+
+RTC_NOINIT_ATTR static int crash_count = 0;
+
+bool is_crash_mode() {
+  return crash_count > MAX_CRASH_COUNT;
+}
+
+void init_crash_detection() {
+  // crash reason is not sw reset, so not a crash!
+  if(rtc_get_reset_reason(0) != 12) {
+    crash_count = 0;
+  } else {
+    ++crash_count;
+  }
+}
+
+int get_crash_count() {
+  return crash_count;
+}
+
+
+void restart_esp() {
+  --crash_count;
```

Comment:
Does this not have to potential to end up negative and thus making the crash counter go on for longer?
Or am i misunderstanding how RTC_NOINIT_ATTR.

If RTC_NOINIT_ATTR does preserve the value i think doing a check if it's already zero would be good so
that the behaviour is consistant

Reply:
Yeah, thought about that. Since it would only take longer to kick in, I ignored it. But you are probably right.

I changed it, so the reset counter resets after 5 minutes of operation. This prevents a "manual" reboot loop for example if you happen to flash a wrong target and the manual reset call is triggered repeatedly due to button presses.